includes/payments/functions.php

```diff
@@ -222,6 +215,10 @@ function edd_delete_purchase( $payment_id = 0, $update_customer = true, $delete_
 		}
 	}
 
+	if ( $customer && $customer->id && $update_customer ) {
+		$customer->recalculate_stats();
+	}
+
```

Comment:
Again, I wonder if this should have been down here the entire time, and was causing problems in 2.x.

Good catch.